plugins/auth-node/src/IdentityClient.ts

```diff
@@ -67,91 +72,41 @@ export class IdentityClient {
     if (!token) {
       throw new AuthenticationError('No token specified');
     }
-    // Get signing key matching token
-    const key = await this.getKey(token);
-    if (!key) {
-      throw new AuthenticationError('No signing key matching token found');
+    // Check if the keystore needs to be updated
+    const url = await this.discovery.getBaseUrl('auth');
+    const endpoint = new URL(`${url}/.well-known/jwks.json`);
+    if (endpoint !== this.endpoint) {
```

Comment:
This might lead to us throwing away the store way more often than needed. For example a `getBaseUrl` implementation that implements round robin towards a couple of instances with different URLs. We'll need to be able to adapt to new endpoints without losing the existing keys

Reply:
Restored the previous implementation, which will check for an updated URL if the authentication fails and the key has been issued within 10s of the last keystore refresh